covered solution with tests

fn1z opened the pull request.

Commit:
covered solution with tests

sarakhmen commented on src/test/java/core/basesyntax/dao/FileIoTest.java:
```diff
@@ -0,0 +1,46 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+class FileIoTest {
```
you should have single unit test file per class. So, for writer and reader there should be 2 tests

sarakhmen commented on src/test/java/core/basesyntax/model/OperationTest.java:
```diff
@@ -0,0 +1,32 @@
+
+import org.junit.jupiter.api.Test;
+
+class OperationTest {
```
no need to test this enum at all

sarakhmen commented on src/test/java/core/basesyntax/service/operation/OperationHandlersTest.java:
```diff
@@ -0,0 +1,50 @@
+    @Test
+    void purchaseShouldDecrease() {
+        storage.put("apple", 10);
+        OperationHandler purchase = new PurchaseOperation(storage);
```
you should have separate test classes per different operation. Also pay attention to your methods naming convention

sarakhmen replied:
naming convention should be the same across all tests

sarakhmen commented on src/test/java/core/basesyntax/service/DataConverterImplTest.java:
```diff
@@ -0,0 +1,74 @@
+        // PURCHASE
+        assertEquals(Operation.PURCHASE, transactions.get(2).getOperation());
+        assertEquals("orange", transactions.get(2).getFruit());
+        assertEquals(20, transactions.get(2).getQuantity());
```
create expected list of transactions and compare it with actual ones in 1-2 lines of code

sarakhmen reviewed (changes requested):
let's improve your solution :)

Commit:
divided FileIoTest into FileReaderTest & FileWriterTest
deleted OperationTest
OperationHandlersTest deleted and created separate test classes for each operation
DataConverterImplTest fixed

Commit:
divided FileIoTest into FileReaderTest & FileWriterTest
deleted OperationTest
OperationHandlersTest deleted and created separate test classes for each operation
DataConverterImplTest fixed
Overridden equals & hashCode in FruitTransaction

sarakhmen commented on src/test/java/core/basesyntax/dao/FileReaderTest.java:
```diff
@@ -0,0 +1,36 @@
+    @Test
+    void readShouldReturnLines_whenFileExists() {
+        FileWriter writer = new FileWriterImpl();
+        FileReader reader = new FileReaderImpl();
```
You shouldn't create an instance of FileWriter for unit test for FileReader class. Use some native java classes to write data or create a file with corresponding data manually. Also, use @BeforeAll/BeforeEach method for creating instances of the classes that are being tested, so that no duplication is present across different methods. Fix same issue in all places

sarakhmen commented on src/test/java/core/basesyntax/dao/FileReaderTest.java:
```diff
@@ -0,0 +1,36 @@
+        FileWriter writer = new FileWriterImpl();
+        FileReader reader = new FileReaderImpl();
+
+        Path file = tempDir.resolve("input.csv");
```
for files use test resources folder and reference files from that place (src/test/resources)

sarakhmen commented on src/test/java/core/basesyntax/dao/FileReaderTest.java:
```diff
@@ -0,0 +1,36 @@
+        FileReader reader = new FileReaderImpl();
+
+        Path file = tempDir.resolve("input.csv");
+        String data = "type,fruit,quantity\nb,banana,2\n";
```
use system.lineSeparator instead of '\n'. Fix same issue in all places

sarakhmen commented on src/test/java/core/basesyntax/dao/FileReaderTest.java:
```diff
@@ -0,0 +1,36 @@
+        writer.write(data, file.toString());
+
+        List<String> lines = reader.read(file.toString());
+        assertFalse(lines.isEmpty());
+        assertEquals("type,fruit,quantity", lines.get(0));
```
as I can see you don't test that the file contains b,banana,2 data

sarakhmen commented on src/test/java/core/basesyntax/dao/FileWriterTest.java:
```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.dao;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exception.FruitShopException;
+import java.nio.file.Path;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+class FileWriterTest {
+    @TempDir
+    private Path tempDir;
+
+    @Test
+    void writeShouldThrow_whenDataOrPathIsNull() {
+        FileWriter writer = new FileWriterImpl();
+        assertThrows(FruitShopException.class, () -> writer.write(null, null));
+    }
```
this test is too simple. Add additional test methods, at least one for success scenario

sarakhmen commented on src/test/java/core/basesyntax/service/ReportGeneratorImplTest.java:
```diff
@@ -0,0 +1,21 @@
+        String report = rg.getReport();
+        assertTrue(report.startsWith("fruit,quantity"));
+        assertTrue(report.contains("banana,2"));
+        assertTrue(report.contains("apple,1"));
```
compare the result report is the expected content in 1 operation

sarakhmen commented on src/test/java/core/basesyntax/storage/StorageTest.java:
```diff
@@ -0,0 +1,48 @@
+import org.junit.jupiter.api.Test;
+
+class StorageTest {
+    private Storage storage;
```
no need to have a test for this class. You can remove it

sarakhmen reviewed (changes requested):
let's improve your solution one more time :)

Commit:
fixed FileReaderTest
fixed FileWriterTest
fixed ReportGeneratorImplTest
removed StorageTest class & storage folder

Commit:
fixed FileReaderTest
fixed FileWriterTest
fixed ReportGeneratorImplTest
removed StorageTest class & storage folder

Elena-Bruyako commented on src/main/java/core/basesyntax/Main.java:
```diff
@@ -0,0 +1,65 @@
+
+        List<FruitTransaction> transactions = dataConverter.convertToTransaction(inputReport);
+
+        for (FruitTransaction transaction : transactions) {
```
it should be in service, not in Main method

Elena-Bruyako commented on src/main/java/core/basesyntax/Main.java:
```diff
@@ -0,0 +1,65 @@
+        FileWriter fileWriter = new FileWriterImpl();
+        fileWriter.write(resultingReport, "finalReport.csv");
+
+        System.out.println("Отчёт успешно сгенерирован и записан в finalReport.csv");
```
why russian?

Elena-Bruyako commented on src/main/java/core/basesyntax/Main.java:
```diff
@@ -0,0 +1,65 @@
+        String resultingReport = reportGenerator.getReport();
+
+        FileWriter fileWriter = new FileWriterImpl();
+        fileWriter.write(resultingReport, "finalReport.csv");
```
create constant for file path

Elena-Bruyako commented on src/main/java/core/basesyntax/Main.java:
```diff
@@ -0,0 +1,65 @@
+
+public class Main {
+    public static void main(String[] args) {
+        if (args.length < 2) {
```
see example of main method in task description

Elena-Bruyako reviewed (changes requested):
see comments

Commit:
main should be fixed
